source/codegen/service_helpers.py

```diff
@@ -719,8 +719,8 @@ def get_streaming_type(parameters) -> str:
 
 def get_size_param_name(streaming_param) -> str:
     """Get the size parameter name for the given streaming parameter."""
-    if common_helpers.is_array(streaming_param["type"]):
-        return streaming_param["size"]["value"]
+    if common_helpers.is_array(streaming_param["type"]) and streaming_param['direction'] == 'out':
```

Comment:
add more description on the method on why the second condition is needed. As the size would only be presented for read arrays which are out direction. It would make it clearer.

Reply:
updated